transmart-core-db/grails-app/services/org/transmartproject/db/clinical/MultidimensionalDataResourceService.groovy

```diff
@@ -582,6 +582,17 @@ class MultidimensionalDataResourceService implements MultiDimensionalDataResourc
         queryResult
     }
     
+    @Override List<QueryResult> findPatientSets(User user) {
+        List<QueryResult>  queryResultList = []
+        if(((DbUser) user).admin){
+            queryResultList = QtQueryResultInstance.findAll()
+        } else {
+            List<QtQueryInstance> queryInstanceList = QtQueryInstance.findAllByUserId(user.username)
+            queryResultList = QtQueryResultInstance.findAllByQueryInstanceInList(queryInstanceList)
```

Comment:
If you do this with a subquery / criteria query it can be done in 1 query instead of two.